operator/src/main/java/org/keycloak/operator/KeycloakDeployment.java

```diff
@@ -0,0 +1,146 @@
+        // TODO this should be done through an informer to leverage caches
+        // WORKAROUND for: https://github.com/java-operator-sdk/java-operator-sdk/issues/781
+        return client
+                .apps()
+                .deployments()
+                .inNamespace(keycloak.getMetadata().getNamespace())
+                .list()
+                .getItems()
+                .stream()
+                .filter((d) -> d.getMetadata().getName().equals(org.keycloak.operator.Constants.NAME))
+                .findFirst()
+                .orElse(null);
+//                .withName(Constants.NAME)
+//                .get();
+    }
+
+    public void createKeycloakDeployment(Keycloak keycloak) {
+        client
+            .apps()
+            .deployments()
+            .inNamespace(keycloak.getMetadata().getNamespace())
+            .create(newKeycloakDeployment(keycloak));
+    }
+
+    public Deployment newKeycloakDeployment(Keycloak keycloak) {
+        return new DeploymentBuilder()
+                .withNewMetadata()
+                    .withName(Constants.NAME)
+                    .withNamespace(keycloak.getMetadata().getNamespace())
+                    .addToLabels(Constants.MANAGED_BY_LABEL, Constants.MANAGED_BY_VALUE)
+                    .addNewOwnerReference()
+                        .withApiVersion(Constants.CRDS_VERSION)
+                        .withKind(keycloak.getKind())
+                        .withName(keycloak.getMetadata().getName())
+                        .withUid(keycloak.getMetadata().getUid())
+                    .endOwnerReference()
+                .endMetadata()
+
+                .withNewSpec()
+
+                .withNewTemplate()
+                .withNewMetadata()
+                .addToLabels(Constants.DEFAULT_LABELS)
+                .endMetadata()
+                .withNewSpec()
+
+                .addNewInitContainer()
+                .withName("init-container")
+                .withImage(Constants.DEFAULT_KEYCLOAK_INIT_IMAGE)
+                .endInitContainer()
+
+                .addNewContainer()
+                .withName(Constants.NAME)
+                .withImage(Constants.DEFAULT_KEYCLOAK_IMAGE)
+                .addNewPort()
+                .withContainerPort(8443)
+                .withProtocol("TCP")
+                .endPort()
+                .endContainer()
+                .endSpec()
+
+                .endTemplate()
+                .withReplicas(keycloak.getSpec().getInstances())
+                .withNewSelector()
+                .addToMatchLabels(Constants.DEFAULT_LABELS)
+                .endSelector()
+
+                .endSpec()
+                .build();
+    }
+
+    public KeycloakStatus getNextStatus(KeycloakSpec desired, KeycloakStatus prev, Deployment current) {
+
+        var isReady = (current != null &&
+                current.getStatus() != null &&
+                current.getStatus().getReadyReplicas() != null);
```

Comment:
Should also check the desired replicas.

Reply:
yes, status needs to be re-worked anyhow

Reply:
@andreaTP Could you please mark this as resolved?